client/hwlib/src/helpers.rs

```diff
@@ -16,14 +16,48 @@
  *        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
  */
 
+use anyhow::{anyhow, Context, Result};
 use std::path::PathBuf;
+use time::Date;
 
 pub(crate) fn append_to_pathbuf(p: PathBuf, s: &str) -> PathBuf {
     let mut p = p.into_os_string();
     p.push(s);
     p.into()
 }
 
+pub(crate) fn parse_date(raw_date: &str) -> Result<Date> {
```

Comment:
I see that the date in `bios.release_date` is formatted as `yyyy-mm-dd`, but this function only handles dates of a slightly different form. Perhaps to make it clearer which places this function should and shouldn't be used, it could be called something like  `parse_smbios_date`?